### DISCUSSION TITLE
Gebruik referentie implementaties

### DISCUSSION THREAD

**Discussion by michielverhoef** [VNG-Realisatie/gemma-zaken#2325] [Polls]
Zoals u ongetwijfeld weet worden de referentie implementaties gebruikt om de testscripts te ontwikkelen die op het API Test Platform gebruikt worden. Maar dat is niet het enige waarvoor de referentie implementaties gebruikt kunnen worden. Ander gebruiksmogelijkheden zijn bijvoorbeeld:

- Playground of sandbox om het voorgeschreven gedrag te demonstreren.
- Invulling geven aan essentiële componenten (stub) die door een partij niet zelf gerealiseerd worden maar wel nodig zijn voor een correcte implementatie van de API standaard.
- Testcomponent bij implementaties.
- …

We zijn benieuwd waarvoor u de referentie implementaties gebruikt of wilt gaan gebruiken.  Met onderstaande enquête willen we dit in beeld krijgen teneinde te kunnen bepalen of de huidige referentie implementaties voorzien in een behoefte of, indien die niet het geval is, hoe we dit dan wel/beter kunnen doen.

**Comment by joeribekker** [comment]
Namens [Maykin](https://maykinmedia.nl/nl/): Ik twijfelde tussen het antwoord "ik gebruik [het] niet" en "anders".

De reden waarom we het **niet** gebruiken is omdat we (logischerwijs, als bouwers van) Open Zaak inzetten voor zowel test als productie doeleinden. Omdat Open Zaak als hoofddoel heeft de standaard te implementeren is dit een alternatief voor de referentie implementaties.

Echter, we hebben de referentie implementatie recent gebruikt om te zien of iets ([adres-typen zijn mutually exclusive](https://github.com/VNG-Realisatie/gemma-zaken/issues/2297)) wat niet expliciet in de documentatie/specificatie stond, wel in de broncode te zien was. Uiteindelijk weet ik niet of dit nu echt meerwaarde heeft geboden, want we hebben er toch een ticket voor aangemaakt.

**Comment by EduardWitteveen** [comment]
Vanuit Open Zaakbrug heb ik behoefte aan een server-end die ik kan configureren als default zaaksysteem. Wanneer een gebruiker van Open Zaakbrug dat het product wil uitproberen is er een echt zgw-zaaksysteem welke ervoor zorgt dat de Open Zaakbrug kan worden uitgeprobeerd

**Comment by michielverhoef** [comment]
Bedankt voor jullie bijdragen.  Dit geeft meer duidelijkheid.  Ondanks dat ik blij ben met de reacties loopt het niet storm. Daarom sluit ik de enquete. Mochten iemand aanvullend nog willen reageren kan dat altijd natuurlijk.
